cerk/src/kernel/broker_event.rs

```diff
@@ -41,26 +53,84 @@ pub enum BrokerEvent {
     ///
     ConfigUpdated(Config, InternalServerId),
 
-    /// The IncommingCloudEvent event indicates to the receiver that a new CloudEvent has been received from the outside world.
+    /// The IncomingCloudEvent event indicates to the receiver that a new CloudEvent has been received from the outside world.
     /// The event is produced by an input port and is sent to the Kernel. The Kernel sends the same event to the router.
     ///
     /// # Arguments
     ///
     /// * `InternalServerId` - id of the component which received the CloudEvent
+    /// * `CloudEventMessageRoutingId` - the unique identifier of the CloudEvent routing attempt
     /// * `CloudEvent` - the deserialized CloudEvent that the component has received
+    /// * `CloudEventRoutingArgs` - routing arguments to define how a CloudEvent should be routed
+    ///
+    IncomingCloudEvent(
+        InternalServerId,
+        CloudEventMessageRoutingId,
+        CloudEvent,
+        CloudEventRoutingArgs,
+    ),
+
+    /// The `RoutingResult` is the result of a routing from one `IncomingCloudEvent`.
+    /// The event is sent from the router to the kernel and there forwarded as `OutgoingCloudEvent` to the ports.
     ///
-    IncommingCloudEvent(InternalServerId, CloudEvent),
+    /// The `RoutingResult` doesn't have to be sent if there are no destinations (`Vec<OutgoingCloudEvent>.len() == 0`)
+    ///
+    /// # Arguments
+    ///
+    /// * `CloudEventMessageRoutingId` - the unique identifier of the CloudEvent routing attempt
+    /// * `InternalServerId` - the id of the component that received the event from the outside world
+    /// * `Vec<OutgoingCloudEvent>` - the list of events that should be forwarded to the outgoing ports. This vec must only contain `BrokerEvent::OutgoingCloudEvent`. `CloudEventMessageRoutingId` in every message has to be the same as the main `CloudEventMessageRoutingId`.
+    /// * `CloudEventRoutingArgs` - routing arguments to define how a CloudEvent should be routed - this config is used by the kernel; the args for the ports are inside the `Vec<OutgoingCloudEvent>`
+    ///
+    RoutingResult(
+        CloudEventMessageRoutingId,
+        InternalServerId,
+        Vec<BrokerEvent>,
+        CloudEventRoutingArgs,
```

Comment:
would it make sense to put these fields into a struct? I think it would be easier to work with later on (I had to check the docstrings a lot).
```
struct RoutingResultPayload {
  pub message_routing_id: CloudEventMessageRoutingId,
  pub event: CloudEvent,
  pub source_id: InternalServerId,
  pub destination_ids: Vec<InternalServerId>,
  pub routing_args: CloudEventRoutingArgs,
}
```

By moving the cloning of the cloud event from the router to the kernel, we could avoid having the generic `BrokerEvent` in the message (fewer checks if it's the right type of `BrokerEvent`)